scripts/Muon/GUI/ElementalAnalysis2/context/context.py

```diff
@@ -18,6 +19,9 @@
 
 class EAContextADSObserver(MuonContextADSObserver):
 
+    def __init__(self, delete_callback, clear_callback, replace_callback):
```

Comment:
why do we need a class for an observer?

Reply:
The MuonContextADSObserver does not call the delete callback when a workspace group is deleted but that functionality is required so deleteHandle was redefined

Reply:
couldn't you just change the slot for the observer, so it will call what you need?